### ISSUE TITLE
How to link with the quazip and qscintilla system libraries

### ISSUE THREAD

**Issue by blackcross86** [trikset/trik-studio#1632]
RosaLinux
version 2022.2

How to link system libraries to avoid re-assembly of quazip and qscintilla.The project is being assembled, but it's getting in the way.

**Comment by iakov** [comment]
TL;DR: no chance in 2022.2, but some improvements are in `master` already, I will kindly appreciate your feedback.

Full answer.
By some historical reasons (cross-platformness problems, multiplied by obsolete macOSes and Win32 targets) TS uses it's own versions of these libraries.
`QuaZip` has improved, and now there is `CONFIG+=use_system_quazip` qmake option that can be used. Please, check if this works fine for ROSA. We are ready to improve TS build for Linux distros. Feel free to open a separate ROSA-build related issue. 

`QScintilla` problem is different. Firstly, TS contains a patched version of QScintilla. Probably, these patches are useless after QScintilla is updated. However, until TS updates to the latest QScintilla, it is close to impossible to state this. Secondly, there was an attempt to introduce system QScintilla, and I've refreshed it (follow #1621, maybe it will be merged soon).

**Comment by blackcross86** [comment]
```patch
diff --git a/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-common.sh b/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-common.sh
index 0d04a7f..733222b 100755
--- a/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-common.sh
+++ b/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-common.sh
@@ -11,8 +11,6 @@ rsync -a "$INSTALLER_ROOT"/licenses/OFL*.txt                                 "$P
 rsync -a "$INSTALLER_ROOT"/licenses/QsKineticScroller-license*.txt           "$PWD"
 rsync -a "$INSTALLER_ROOT"/licenses/QsLog-license*.txt                       "$PWD"
 rsync -a "$INSTALLER_ROOT"/licenses/QtPropertyBrowser-license*.txt           "$PWD"
-rsync -a "$INSTALLER_ROOT"/licenses/QScintilla-GPL-exception*.txt            "$PWD"
-rsync -a "$INSTALLER_ROOT"/licenses/Scintilla-license*.txt                   "$PWD"
 rsync -a "$INSTALLER_ROOT"/licenses/WindowsModernStyle-license*.txt          "$PWD"
 rsync -a "$INSTALLER_ROOT"/licenses/DejaVu_Fonts_License*.txt                "$PWD"
 rsync -a "$INSTALLER_ROOT"/licenses/acknowledgements.txt                     "$PWD"/../data
diff --git a/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-linux-gnu.sh b/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-linux-gnu.sh
index 1590539..f5e811b 100755
--- a/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-linux-gnu.sh
+++ b/installer/packages/qreal-base/ru.qreal.root/meta/prebuild-linux-gnu.sh
@@ -20,7 +20,6 @@ rsync -a "$BIN_DIR"/libqrgraph.so*                                          "$PW
 rsync -a "$BIN_DIR"/libqrrepo.so*                                           "$PWD"/../data/lib/
 rsync -a "$BIN_DIR"/libqrutils.so*                                          "$PWD"/../data/lib/
 rsync -a "$BIN_DIR"/libqrtext.so*                                           "$PWD"/../data/lib/
-rsync -a "$BIN_DIR"/libqscintilla2.so*                                      "$PWD"/../data/lib/
 rsync -a "$BIN_DIR"/libtrikQsLog.so*                                        "$PWD"/../data/lib/
 rsync -a "$BIN_DIR"/libqrgui-brand-manager.so*                              "$PWD"/../data/lib/
 rsync -a "$BIN_DIR"/libqrgui-controller.so*                                 "$PWD"/../data/lib/
diff --git a/plugins/robots/checker/scripts/build-checker-installer.sh b/plugins/robots/checker/scripts/build-checker-installer.sh
index cfa973d..b6a873d 100755
--- a/plugins/robots/checker/scripts/build-checker-installer.sh
+++ b/plugins/robots/checker/scripts/build-checker-installer.sh
@@ -123,7 +123,6 @@ $qRealDir/libqrkernel.so* \
 $qRealDir/libqrrepo.so* \
 $qRealDir/libqrtext.so* \
 $qRealDir/libqrutils.so* \
-$qRealDir/libqscintilla2.so* \
 $qRealDir/libqrtext.so* .
 
 # Copying TRIKStudio plugins
diff --git a/plugins/robots/generators/ev3/ev3RbfGenerator/ev3RbfGenerator.pro b/plugins/robots/generators/ev3/ev3RbfGenerator/ev3RbfGenerator.pro
index 60ad1df..3023dac 100644
--- a/plugins/robots/generators/ev3/ev3RbfGenerator/ev3RbfGenerator.pro
+++ b/plugins/robots/generators/ev3/ev3RbfGenerator/ev3RbfGenerator.pro
@@ -32,7 +32,7 @@ includes(plugins/robots/generators/ev3/ev3GeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 qrgui-text-editor qrtext \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 qrgui-text-editor qrtext \
 		robots-generator-base robots-ev3-generator-base robots-ev3-kit robots-kit-base robots-utils)
 
 TRANSLATIONS = $$PWD/../../../../../qrtranslations/ru/plugins/robots/ev3RbfGenerator_ru.ts
diff --git a/plugins/robots/generators/nxt/nxtOsekCGenerator/nxtOsekCGenerator.pro b/plugins/robots/generators/nxt/nxtOsekCGenerator/nxtOsekCGenerator.pro
index 798a25a..1cde20e 100644
--- a/plugins/robots/generators/nxt/nxtOsekCGenerator/nxtOsekCGenerator.pro
+++ b/plugins/robots/generators/nxt/nxtOsekCGenerator/nxtOsekCGenerator.pro
@@ -31,7 +31,7 @@ includes(plugins/robots/generators/nxt/nxtGeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-utils robots-nxt-kit \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-utils robots-nxt-kit \
 		robots-generator-base robots-nxt-generator-base)
 
 TRANSLATIONS = \
diff --git a/plugins/robots/generators/nxt/nxtRussianCGenerator/nxtRussianCGenerator.pro b/plugins/robots/generators/nxt/nxtRussianCGenerator/nxtRussianCGenerator.pro
index 9fec79d..8bb7da7 100644
--- a/plugins/robots/generators/nxt/nxtRussianCGenerator/nxtRussianCGenerator.pro
+++ b/plugins/robots/generators/nxt/nxtRussianCGenerator/nxtRussianCGenerator.pro
@@ -29,7 +29,7 @@ includes(plugins/robots/generators/nxt/nxtGeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-nxt-generator-base)
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-nxt-generator-base)
 
 TRANSLATIONS = \
 	$$PWD/../../../../../qrtranslations/ru/plugins/robots/nxtRussianCGenerator_ru.ts \
diff --git a/plugins/robots/generators/pioneer/pioneerLuaGenerator/pioneerLuaGenerator.pro b/plugins/robots/generators/pioneer/pioneerLuaGenerator/pioneerLuaGenerator.pro
index 788b191..1845c2e 100644
--- a/plugins/robots/generators/pioneer/pioneerLuaGenerator/pioneerLuaGenerator.pro
+++ b/plugins/robots/generators/pioneer/pioneerLuaGenerator/pioneerLuaGenerator.pro
@@ -31,7 +31,7 @@ includes( \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 qrgui-text-editor qrgui-preferences-dialog qrtext robots-kit-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 qrgui-text-editor qrgui-preferences-dialog qrtext robots-kit-base \
 		robots-generator-base robots-pioneer-kit robots-utils)
 
 TRANSLATIONS = $$PWD/../../../../../qrtranslations/ru/plugins/robots/pioneerLuaGenerator_ru.ts
diff --git a/plugins/robots/generators/trik/trikFSharpGeneratorLibrary/trikFSharpGeneratorLibrary.pro b/plugins/robots/generators/trik/trikFSharpGeneratorLibrary/trikFSharpGeneratorLibrary.pro
index 2f3f9cb..e51a8b7 100644
--- a/plugins/robots/generators/trik/trikFSharpGeneratorLibrary/trikFSharpGeneratorLibrary.pro
+++ b/plugins/robots/generators/trik/trikFSharpGeneratorLibrary/trikFSharpGeneratorLibrary.pro
@@ -30,7 +30,7 @@ includes( \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base robots-utils \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base robots-utils \
 		qrgui-preferences-dialog robots-kit-base \
 )
 
diff --git a/plugins/robots/generators/trik/trikPascalABCGeneratorLibrary/trikPascalABCGeneratorLibrary.pro b/plugins/robots/generators/trik/trikPascalABCGeneratorLibrary/trikPascalABCGeneratorLibrary.pro
index c1a6c14..66b874b 100644
--- a/plugins/robots/generators/trik/trikPascalABCGeneratorLibrary/trikPascalABCGeneratorLibrary.pro
+++ b/plugins/robots/generators/trik/trikPascalABCGeneratorLibrary/trikPascalABCGeneratorLibrary.pro
@@ -28,7 +28,7 @@ includes(plugins/robots/generators/trik/trikGeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils qrgui-preferences-dialog \
 )
 
diff --git a/plugins/robots/generators/trik/trikPythonGeneratorLibrary/trikPythonGeneratorLibrary.pro b/plugins/robots/generators/trik/trikPythonGeneratorLibrary/trikPythonGeneratorLibrary.pro
index bb4e731..a472c03 100644
--- a/plugins/robots/generators/trik/trikPythonGeneratorLibrary/trikPythonGeneratorLibrary.pro
+++ b/plugins/robots/generators/trik/trikPythonGeneratorLibrary/trikPythonGeneratorLibrary.pro
@@ -28,7 +28,7 @@ includes(plugins/robots/generators/trik/trikGeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils qrgui-text-editor \
 )
 
diff --git a/plugins/robots/generators/trik/trikQtsGeneratorLibrary/trikQtsGeneratorLibrary.pro b/plugins/robots/generators/trik/trikQtsGeneratorLibrary/trikQtsGeneratorLibrary.pro
index 17fac2d..e977f5c 100644
--- a/plugins/robots/generators/trik/trikQtsGeneratorLibrary/trikQtsGeneratorLibrary.pro
+++ b/plugins/robots/generators/trik/trikQtsGeneratorLibrary/trikQtsGeneratorLibrary.pro
@@ -28,7 +28,7 @@ includes(plugins/robots/generators/trik/trikGeneratorBase \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils qrgui-text-editor \
 )
 
diff --git a/plugins/robots/generators/trik/trikV62PythonGenerator/trikV62PythonGenerator.pri b/plugins/robots/generators/trik/trikV62PythonGenerator/trikV62PythonGenerator.pri
index d381290..663dd84 100644
--- a/plugins/robots/generators/trik/trikV62PythonGenerator/trikV62PythonGenerator.pri
+++ b/plugins/robots/generators/trik/trikV62PythonGenerator/trikV62PythonGenerator.pri
@@ -24,7 +24,7 @@ includes( \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils robots-trik-python-generator-library \
 )
 
diff --git a/plugins/robots/generators/trik/trikV62QtsGenerator/trikV62QtsGenerator.pri b/plugins/robots/generators/trik/trikV62QtsGenerator/trikV62QtsGenerator.pri
index 71b851d..4a36ac4 100644
--- a/plugins/robots/generators/trik/trikV62QtsGenerator/trikV62QtsGenerator.pri
+++ b/plugins/robots/generators/trik/trikV62QtsGenerator/trikV62QtsGenerator.pri
@@ -24,7 +24,7 @@ includes( \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils robots-trik-qts-generator-library \
 )
 
diff --git a/plugins/robots/generators/trik/trikV6QtsGenerator/trikV6QtsGenerator.pro b/plugins/robots/generators/trik/trikV6QtsGenerator/trikV6QtsGenerator.pro
index a828a56..46326d5 100644
--- a/plugins/robots/generators/trik/trikV6QtsGenerator/trikV6QtsGenerator.pro
+++ b/plugins/robots/generators/trik/trikV6QtsGenerator/trikV6QtsGenerator.pro
@@ -33,7 +33,7 @@ includes( \
 		qrtext \
 )
 
-links(qrkernel qrutils qrrepo qscintilla2 robots-generator-base robots-trik-generator-base \
+links(qrkernel qrutils qrrepo qscintilla2_qt5 robots-generator-base robots-trik-generator-base \
 		robots-kit-base robots-trik-kit robots-utils robots-trik-qts-generator-library \
 )
 
diff --git a/plugins/robots/interpreters/trikKitInterpreterCommon/trikKitInterpreterCommon.pri b/plugins/robots/interpreters/trikKitInterpreterCommon/trikKitInterpreterCommon.pri
index 8c7ad27..0e425d6 100644
--- a/plugins/robots/interpreters/trikKitInterpreterCommon/trikKitInterpreterCommon.pri
+++ b/plugins/robots/interpreters/trikKitInterpreterCommon/trikKitInterpreterCommon.pri
@@ -27,7 +27,7 @@ includes( \
 		plugins/robots/thirdparty/trikRuntime/trikRuntime/trikScriptRunner \
 		)
 
-links(qrkernel qrutils qrtext qrgui-preferences-dialog qrgui-tool-plugin-interface qscintilla2 \
+links(qrkernel qrutils qrtext qrgui-preferences-dialog qrgui-tool-plugin-interface qscintilla2_qt5 \
                 robots-utils robots-kit-base robots-2d-model robots-trik-kit \
                 trikControl trikNetwork trikScriptRunner trikKernel qrgui-text-editor \
 		)
diff --git a/plugins/tools/exterminatus/exterminatus.pro b/plugins/tools/exterminatus/exterminatus.pro
index c13e682..fc71305 100644
--- a/plugins/tools/exterminatus/exterminatus.pro
+++ b/plugins/tools/exterminatus/exterminatus.pro
@@ -18,7 +18,7 @@ TEMPLATE = lib
 CONFIG += plugin
 DESTDIR = $$DESTDIR/plugins/tools/
 
-includes(qrgui thirdparty/qscintilla/Qt4Qt5)
+includes(qrgui /usr/include/qt5/Qsci)
 
 QT += widgets
 
diff --git a/plugins/tools/refactoring/refactoringSupport/refactoringSupport.pro b/plugins/tools/refactoring/refactoringSupport/refactoringSupport.pro
index 2914156..5157b01 100644
--- a/plugins/tools/refactoring/refactoringSupport/refactoringSupport.pro
+++ b/plugins/tools/refactoring/refactoringSupport/refactoringSupport.pro
@@ -20,7 +20,7 @@ TEMPLATE = lib
 CONFIG += plugin
 DESTDIR = $$DESTDIR/plugins/tools/
 
-includes(thirdparty/qscintilla/Qt4Qt5 qrgui)
+includes(/usr/include/qt5/Qsci qrgui)
 
 links(qrkernel qrutils qrrepo qrgui-preferences-dialog)
 
diff --git a/plugins/tools/uxInfo/uxInfo.pro b/plugins/tools/uxInfo/uxInfo.pro
index b2c88d9..08c29b2 100644
--- a/plugins/tools/uxInfo/uxInfo.pro
+++ b/plugins/tools/uxInfo/uxInfo.pro
@@ -18,7 +18,7 @@ TEMPLATE = lib
 CONFIG += plugin
 DESTDIR = $$DESTDIR/plugins/tools/
 
-includes(thirdparty/qscintilla/Qt4Qt5 qrgui)
+includes(/usr/include/qt5/Qsci qrgui)
 
 links(qrkernel qslog qrgui-preferences-dialog qrgui-tool-plugin-interface)
 
diff --git a/plugins/tools/visualInterpreter/visualInterpreter.pro b/plugins/tools/visualInterpreter/visualInterpreter.pro
index 2a844cf..b221a6a 100644
--- a/plugins/tools/visualInterpreter/visualInterpreter.pro
+++ b/plugins/tools/visualInterpreter/visualInterpreter.pro
@@ -21,7 +21,7 @@ CONFIG += plugin
 
 DESTDIR = $$DESTDIR/plugins/tools/
 
-includes(thirdparty/qscintilla/Qt4Qt5 qrgui qrtext)
+includes(/usr/include/qt5/Qsci qrgui qrtext)
 
 links(qrkernel qrutils qrgui-preferences-dialog)
 
diff --git a/qrgui/mainWindow/mainWindow.cpp b/qrgui/mainWindow/mainWindow.cpp
index 31ef8ce..c84c4b0 100644
--- a/qrgui/mainWindow/mainWindow.cpp
+++ b/qrgui/mainWindow/mainWindow.cpp
@@ -46,8 +46,8 @@
 #include <qrutils/smartDock.h>
 #include <qrutils/graphicsUtils/animatedEffects.h>
 #include <qrutils/xmlUtils.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qsciprinter.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qsciscintillabase.h>
+#include <Qsci/qsciprinter.h>
+#include <Qsci/qsciscintillabase.h>
 
 #include <qrgui/controller/controller.h>
 #include <qrgui/dialogs/findReplaceDialog.h>
diff --git a/qrgui/mainWindow/mainWindow.pro b/qrgui/mainWindow/mainWindow.pro
index 7533605..057b743 100644
--- a/qrgui/mainWindow/mainWindow.pro
+++ b/qrgui/mainWindow/mainWindow.pro
@@ -21,7 +21,7 @@ DEFINES += TRIK_STUDIO_VERSION='\\"$$TRIK_STUDIO_VERSION\\"'
 
 QT += widgets printsupport xml svg
 
-links(qrkernel qrutils qrtext qrrepo qscintilla2 qrgui-models qrgui-editor qrgui-controller qrgui-dialogs qrgui-preferences-dialog \
+links(qrkernel qrutils qrtext qrrepo qscintilla2_qt5 qrgui-models qrgui-editor qrgui-controller qrgui-dialogs qrgui-preferences-dialog \
 		qrgui-text-editor qrgui-mouse-gestures qrgui-hotkey-manager qrgui-brand-manager  \
 		qrgui-facade qrgui-plugin-manager qrgui-tool-plugin-interface qrgui-thirdparty \
 )
diff --git a/qrgui/textEditor/languageInfo.h b/qrgui/textEditor/languageInfo.h
index 189239b..17395c6 100644
--- a/qrgui/textEditor/languageInfo.h
+++ b/qrgui/textEditor/languageInfo.h
@@ -16,11 +16,11 @@
 
 #include <QtCore/QObject>
 
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qscilexerpython.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qscilexercpp.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qscilexerjavascript.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qscilexerpascal.h>
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qscilexerlua.h>
+#include <Qsci/qscilexerpython.h>
+#include <Qsci/qscilexercpp.h>
+#include <Qsci/qscilexerjavascript.h>
+#include <Qsci/qscilexerpascal.h>
+#include <Qsci/qscilexerlua.h>
 
 #include "qrgui/textEditor/textEditorDeclSpec.h"
 
diff --git a/qrgui/textEditor/qscintillaTextEdit.cpp b/qrgui/textEditor/qscintillaTextEdit.cpp
index 66ddead..753a40d 100644
--- a/qrgui/textEditor/qscintillaTextEdit.cpp
+++ b/qrgui/textEditor/qscintillaTextEdit.cpp
@@ -17,7 +17,7 @@
 #include <QtWidgets/QShortcut>
 #include <QtCore/QRegularExpression>
 
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qsciapis.h>
+#include <Qsci/qsciapis.h>
 #include <brandManager/brandManager.h>
 #include <qrutils/widgets/searchLinePanel.h>
 
diff --git a/qrgui/textEditor/qscintillaTextEdit.h b/qrgui/textEditor/qscintillaTextEdit.h
index 26302c7..9949c02 100644
--- a/qrgui/textEditor/qscintillaTextEdit.h
+++ b/qrgui/textEditor/qscintillaTextEdit.h
@@ -16,7 +16,7 @@
 
 #include <QtCore/QPersistentModelIndex>
 
-#include <thirdparty/qscintilla/Qt4Qt5/Qsci/qsciscintilla.h>
+#include <Qsci/qsciscintilla.h>
 #include <qrgui/plugins/toolPluginInterface/usedInterfaces/editorInterface.h>
 
 #include "qrgui/textEditor/textEditorDeclSpec.h"
diff --git a/qrgui/textEditor/textEditor.pro b/qrgui/textEditor/textEditor.pro
index 70d6e02..7109c55 100644
--- a/qrgui/textEditor/textEditor.pro
+++ b/qrgui/textEditor/textEditor.pro
@@ -18,7 +18,7 @@ TEMPLATE = lib
 
 include(../../global.pri)
 
-links(qrkernel qrutils qscintilla2 qrgui-tool-plugin-interface qrgui-brand-manager)
+links(qrkernel qrutils qscintilla2_qt5 qrgui-tool-plugin-interface qrgui-brand-manager)
 
 QT += widgets
 
@@ -32,7 +32,7 @@ INCLUDEPATH = \
 	$$PWD/ \
 	$$PWD/../ \
 	$$PWD/../../ \
-	$$PWD/../../thirdparty/qscintilla/Qt4Qt5 \
+	/usr/include/qt5/Qsci \
 
 HEADERS += \
 	$$PWD/textEditorDeclSpec.h \
diff --git a/qrtest/editorPluginTestingFramework/editorPluginTestingFramework.pro b/qrtest/editorPluginTestingFramework/editorPluginTestingFramework.pro
index b064269..4d5be6c 100644
--- a/qrtest/editorPluginTestingFramework/editorPluginTestingFramework.pro
+++ b/qrtest/editorPluginTestingFramework/editorPluginTestingFramework.pro
@@ -170,7 +170,7 @@ OTHER_FILES += \
 	$$PWD/travisConfigurationParameters.xml \
 	$$PWD/configurationParameters.xml \
 
-links (qrrepo qrtext qrkernel qrutils qrmc qrgui-plugin-manager qrgui-editor qrgui-thirdparty qscintilla2 qrgraph \
+links (qrrepo qrtext qrkernel qrutils qrmc qrgui-plugin-manager qrgui-editor qrgui-thirdparty qscintilla2_qt5 qrgraph \
 		qrgui-tool-plugin-interface qrgui-models qrgui-controller qrgui-dialogs qrgui-preferences-dialog qrgui-facade \
 		qrgui-text-editor qrgui-mouse-gestures qrgui-hotkey-manager qrgui-brand-manager qrgui-meta-meta-model qslog)
 
diff --git a/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin1/mockKitPlugin1.pro b/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin1/mockKitPlugin1.pro
index 463c0fa..23f5b85 100644
--- a/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin1/mockKitPlugin1.pro
+++ b/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin1/mockKitPlugin1.pro
@@ -21,7 +21,7 @@ DESTDIR = $$DESTDIR/plugins/unittests/
 
 QT += widgets
 
-includes(thirdparty/qscintilla/Qt4Qt5 plugins/robots/common/kitBase plugins/robots/utils qrgui qrtext)
+includes(/usr/include/qt5/Qsci plugins/robots/common/kitBase plugins/robots/utils qrgui qrtext)
 
 links(robots-kit-base)
 
diff --git a/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin2/mockKitPlugin2.pro b/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin2/mockKitPlugin2.pro
index 63c4917..e085e28 100644
--- a/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin2/mockKitPlugin2.pro
+++ b/qrtest/unitTests/pluginsTests/robotsTests/interpretersTests/support/mockKitPlugin2/mockKitPlugin2.pro
@@ -21,7 +21,7 @@ DESTDIR = $$DESTDIR/plugins/unittests/
 
 QT += widgets
 
-includes(thirdparty/qscintilla/Qt4Qt5 plugins/robots/common/kitBase plugins/robots/utils qrgui qrtext)
+includes(/usr/include/qt5/Qsci plugins/robots/common/kitBase plugins/robots/utils qrgui qrtext)
 
 links(robots-kit-base)
 
diff --git a/thirdparty/thirdparty.pro b/thirdparty/thirdparty.pro
index 81857c5..99ef0e8 100644
--- a/thirdparty/thirdparty.pro
+++ b/thirdparty/thirdparty.pro
@@ -18,7 +18,6 @@ HEADERS += \
 	runExtensions.h \
 
 SUBDIRS = \
-	qscintilla/Qt4Qt5/qscintilla.pro \
 	qslog/QsLogSharedLibrary.pro \
 	gamepad/gamepad.pro \
 	quazip/quazip.pro \

```

**Comment by iakov** [comment]
Usually, few steps are enough, if corresponding packages were installed in advance. Like `mkdir build && cd build && qmake ../studio/studio.pro CONFIG+=ccache CONFIG+=release CONFIG+=force_debug_info ...` .
If something does not work, let's check exact problems: what you expect, and what is happening.

**Comment by blackcross86** [comment]
> Usually, few steps are enough, if corresponding packages were installed in advance. Like `mkdir build && cd build && qmake ../studio/studio.pro CONFIG+=ccache CONFIG+=release CONFIG+=force_debug_info ...`. If something does not work, let's check exact problems: what you expect, and what is happening.

Everything came together successfully.Starting the installation, where should it all be installed? Example: gamepad is not displayed in the startup menu.

**Comment by iakov** [comment]
External utilities are to be used from `Tools -> External tools` menu, and are configured via `externalTools.xml`. BTW, do not expect `make install` to work as usually, please, it is totally broken.

**Comment by blackcross86** [comment]
> External utilities are to be used from `Tools -> External tools` menu, and are configured via `externalTools.xml`. BTW, do not expect `make install` to work as usually, please, it is totally broken.

make install, as I already understood, all folders and files are scattered. But it is not clear where to install, there are installation directories, what is the location of files in your program for full use?

**Comment by iakov** [comment]
Well... Not a simple question, actually. Please, take a look into the [QtIFw packaged installer](https://dl.trikset.com/ts/fresh/installer/trik-studio-installer-linux-2022.2.run)/installation. Or check AltLinux packaging "hacks". 

TS has patched RPATH, and expects 'lib' folder (AFAIR) and some not obvious folders structure. Since this has to be fixed, please, feel free to mention your distro problems. 

Also, there is a `trik-studio.ini` by QSettings, that can be used to configure paths (for plugins, resources, etc.), probably, `/etc/xdg/.../trik-studio.ini` can be used to override. Take a look at [platform-specific section](https://doc.qt.io/qt-6/qsettings.html#platform-specific-notes) about `/etc/xdg`/`$XDG_CONFIG_DIRS`
This screenshot shows search paths on Ubuntu:
![image](https://user-images.githubusercontent.com/706759/222227371-05ee2fa6-bb0c-4c54-84eb-c4b4c86c3e0b.png)
Thus, this `.ini` can be used to override paths, and the `.ini` itself can be found after the first run in users `.config` folder.

**Comment by blackcross86** [comment]
Altlinux has a crooked project, just like me.But thank you all the same for the explanations.

**Comment by iakov** [comment]
Let's stay in touch for further TS improvements. Thanks to AltLinux, some improvements generally useful for distro packaging have been upstream-ed already.

**Comment by iakov** [comment]
@blackcross86 , 2024.1 is ready. Lot of refactorings and build system / CI improvements in this release, thus you are welcome to report any issues. If you are the maintainer of the corresponding package in any distro, or a amatuer TS user, let's try to impove build system toward your needs together.
Also, I'm interested and ready to take a sight at your fixes, maybe we can help to simplify furter maintanance. I've created [this issue ](https://github.com/trikset/trik-studio/issues/1843) to maintain a list of contributions for futher integration.
FYI: 2024.1 has `CONFIG+=use_system_qscintilla2` and `CONFIG+=use_system_quazip` options for `qmake` . Maybe this helps.
